src/Cake.Core/Diagnostics/Console/AnsiDetector.cs

```diff
@@ -0,0 +1,124 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////
+// Portions of this code was ported from the supports-ansi project by Qingrong Ke
+// https://github.com/keqingrong/supports-ansi/blob/master/index.js
+// Copyright (c) Qingrong Ke. Licensed under the MIT license.
+/////////////////////////////////////////////////////////////////////////////////////////////////////
+
+using System;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Text.RegularExpressions;
+
+namespace Cake.Core.Diagnostics
+{
+    internal static class AnsiDetector
+    {
+        private static readonly Regex[] _regexes;
+
+        static AnsiDetector()
+        {
+            _regexes = new[]
+            {
+                new Regex("^xterm"), // xterm, PuTTY, Mintty
+                new Regex("^rxvt"), // RXVT
+                new Regex("^eterm"), // Eterm
+                new Regex("^screen"), // GNU screen, tmux
+                new Regex("tmux"), // tmux
+                new Regex("^vt100"), // DEC VT series
+                new Regex("^vt102"), // DEC VT series
+                new Regex("^vt220"), // DEC VT series
+                new Regex("^vt320"), // DEC VT series
+                new Regex("ansi"), // ANSI
+                new Regex("scoansi"), // SCO ANSI
+                new Regex("cygwin"), // Cygwin, MinGW
+                new Regex("linux"), // Linux console
+                new Regex("konsole"), // Konsole
+                new Regex("bvterm"), // Bitvise SSH Client
+            };
+        }
+
+        public static bool SupportsAnsi(ICakeEnvironment environment)
+        {
```

Comment:
Github action doesn't setup a correct PTY but does support ansi, so should addd something like.
```suggestion
        {
            if (!string.IsNullOrEmpty(environment.GetEnvironmentVariable("GITHUB_ACTION")))
            {
                return true;
            }
```

Reply:
@devlead Is that the environment variable name or pseudo code?

Reply:
@devlead Nevermind, found it here: https://help.github.com/en/actions/configuring-and-managing-workflows/using-environment-variables#default-environment-variables